pennylane/data/dataset.py

```diff
@@ -116,7 +118,7 @@ def __std_init__(self, data_name, folder, attr_prefix, docstring):
         if not os.path.exists(self._fullfile):
             self._fullfile = None
 
-        for f in glob(f"{self._folder}{os.path.sep}*.dat"):
+        for f in glob(self._prefix.format("*")):
```

Comment:
not quite related, I just realized while testing that this used to read files that didn't have your prefix but were in your folder (because the assumption is that there shouldn't be any but I made one while testing just for funzies)

Reply:
Not sure if I understand what you mean by your comment here. 🤔